ban/commands/ignsna.py

```diff
@@ -0,0 +1,48 @@
+import os
+from ban.commands import command, report
+from ban.core.models import (Municipality, PostCode)
+from .helpers import session, batch, nodiff
+
+__namespace__ = 'import'
+
+
+@command
+@nodiff
+def ignsna(path, **kwargs):
+    """Import from IGN/Laposte BDUNI
+    :param path: directory location of the files hsp7aaaa.ai, hsv7aaaa.ai and hsw4aaaa.ai"""
+
+    municipality_zipcode_file = os.path.join(path, 'hsp7aaaa.ai')
+    if municipality_zipcode_file is not None:
+        lines = open(municipality_zipcode_file).readlines()
+        batch(process_municipality_file, lines, max_value=len(lines))
+
+
+@session
+def process_municipality_file(line):
+    if line[50] == 'M':
+        insee = line[6:11]
+        post_code = line[89:94]
+        version = 1
+        try:
+            municipality = Municipality.get(Municipality.insee == insee)
+        except Municipality.DoesNotExist:
+            return report('Municipality Not Existing', insee, report.WARNING)
+        if municipality:
+            try:
+                postcode = PostCode.get(PostCode.code == post_code)
+            except PostCode.DoesNotExist:
+                postcode = None
+                data = dict(code=post_code, version=version)
+                validator = PostCode.validator(instance=postcode, **data)
```

Comment:
Also, no need to make a temporary dict, given that you have few arguments and they are used only once:
```
validator = PostCode.validator(code=post_code, version=version)
```